### ISSUE TITLE
About how to update anchors

### ISSUE THREAD

**Issue by Wenlong0913** [ultralytics/yolov5#5007]
## ❔Question
I'd like to ask how to update anchors.
I'm trying to train a new model based on pretrained model. The program output the following information:

autoanchor: Running kmeans for x anchors on xxx points...
autoanchor: thr=x: x best possible recall, x anchors past thr
autoanchor: n=9, img_size=x, metric_all=x/x-mean/best, past_thr=x-mean: x,x,  x,x,  x,x,  x,x,  x,x,  x,x,  x,x,  x,x,  x,x
autoanchor: Evolving anchors with Genetic Algorithm:: 100% 1000/1000 [10:36<00:00,  1.57it/s]
autoanchor: thr=x: x  best possible recall, x anchors past thr
autoanchor: n=9, img_size=x, metric_all=x/x-mean/best, past_thr=x-mean: x0,x0 ,x1,x1,  x2,x2,  x3,x3,  x4,x4,  x5,x5,  x6,x6,  x7,x7,  x7,x7,  x8,x8

Should we update the model.yaml by replacing the anchors section? 

## Additional context
Take modelv5m.yaml as example.

```
#anchors:
#  - [10,13, 16,30, 33,23]  # P3/8
#  - [30,61, 62,45, 59,119]  # P4/16
#  - [116,90, 156,198, 373,326]  # P5/32
 
anchors:
  - [x0,x0, x1,x1, x2,x2]  # P3/8
  - [x3,x3, x4,x4, x5,x5]  # P4/16
  - [x6,x6, x7,x7, x8,x8]  # P5/32
```

Is that correct? Is there any other places we should update?

Any info will be greatly appreciated.

**Comment by glenn-jocher** [comment]
@Wenlong0913 yes you can update your yaml as shown if you want to. 

No action is required from you, autoanchor automatically places new anchors in your model.

**Comment by Wenlong0913** [comment]
Thank you. Yeah, you're right. The reason that I intent to update anchor manually is a NCCL error.
I tried to train objects365 from scratch. I tried 50 epochs, everything looks good to me. Then, I use hyp.finetune_objects365.yaml for fine tuning the model for the fastest conversions as #3392 mentioned. 
`python -m torch.distributed.launch --nproc_per_node 2 train.py --data obj365.yml --cfg models/yolov5m.yaml --weights runs/train/exp/best.pt --hyp hyp.finetune_objects365.yaml`
The program will run the GE algo and update anchors before the training start.
However, it failed and restarted the workgroup. Here is the output info.
```
what():  [Rank 1] Watchdog caught collective operation timeout: WorkNCCL(OpType=ALLREDUCE, Timeout(ms)=60000) ran for 67438 milliseconds before timing out.
[E ProcessGroupNCCL.cpp:327] Some NCCL operations have failed or timed out. Due to the asynchronous nature of CUDA kernels, subsequent GPU operations might run on corrupted/incomplete data. To avoid this inconsistency, we are taking the entire process down.
terminate called after throwing an instance of 'std::runtime_error'
```
If I only run it on single gpu, the above error disappears. I have no idea how to avoid that error, and I notice the message output by the program. 
```New anchors saved to model. Update model *.yaml to use these anchors in the future.```
So here I am, asking about how to update the anchors. If there is something wrong with train command, or any suggestions regard the NCCL error, please let me know.

Any help will be greatly appreciated.

**Comment by glenn-jocher** [comment]
@Wenlong0913 👋 Hello! Thanks for asking about **resuming training**. YOLOv5 🚀 Learning Rate (LR) [schedulers](https://github.com/ultralytics/yolov5/blob/9f3a388cea66ca71b9bbe057929eed8fbc920a7e/train.py#L109-L142) follow predefined LR curves for the fixed number of `--epochs` defined at training start (`default=300`), and are designed to fall to a minimum LR on the final epoch for best training results. For this reason you can not modify the number of epochs once training has started.

<img width="850" alt="LR Curves" src="https://user-images.githubusercontent.com/26833433/119332195-8e113a80-bc88-11eb-8eb7-7494ff1d5c01.png">

If your training was **interrupted** for any reason you may continue where you left off using the `--resume` argument. If your training fully completed, you can start a **new** training from any model using the `--weights` argument. Examples:

## Resume Single-GPU

You may not change settings when resuming, and **no additional arguments** other than `--resume` should be passed, with an optional path to the checkpoint you'd like to resume from. If no checkpoint is passed the most recently updated `last.pt` in your `yolov5/` directory is automatically found and used:
```bash
python train.py --resume  # automatically find latest checkpoint (searches yolov5/ directory)
python train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Resume Multi-GPU

[Multi-GPU DDP trainings](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training) must be resumed with the same GPUs and DDP command, i.e. assuming 8 GPUs:
```bash
python -m torch.distributed.run --nproc_per_node 8 train.py --resume  # resume latest checkpoint
python -m torch.distributed.run --nproc_per_node 8 train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Start from Pretrained

If you would like to **start training** from a fully trained model, use the `--weights` argument, not the `--resume` argument:
```bash
python train.py --weights path/to/best.pt  # start from pretrained model
```

Good luck and let us know if you have any other questions!
